File: FitAppAPI/FitAppAPI/Models/TrainingExercise.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FitAppAPI.Models {
    public class TrainingExercise
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int TrainingId { get; set; }

        public Training Training { get; set; }

        [Required]
        public int ExerciseId { get; set; }

        public Exercise Exercise { get; set; }

        [Required]
        public int Sets { get; set; } // Подходы

        [Required]
        public int Reps { get; set; } // Повторения

        public decimal Weight { get; set; } // Вес

        public int OrderIndex { get; set; } // Порядковый номер
    }
}

File: urbanfit/lib/models/training_exercise.dart
class TrainingExercise {
  final int id;
  final int exerciseId;
  final String? exerciseName;
  final int sets;
  final int reps;
  final double weight;
  final int orderIndex;
  final String? imageUrl;

  TrainingExercise({
    required this.id,
    required this.exerciseId,
    this.exerciseName,
    this.sets = 3,
    this.reps = 10,
    this.weight = 0,
    this.orderIndex = 0,

    this.imageUrl,
  });

  factory TrainingExercise.fromJson(Map<String, dynamic> json) {
    return TrainingExercise(
      id: json['id'],
      exerciseId: json['exerciseId'],
      exerciseName: json['exerciseName'],
      sets: json['sets'],
      reps: json['reps'],
      weight: json['weight']?.toDouble() ?? 0,
      orderIndex: json['orderIndex'],
      imageUrl: json['imageUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'exerciseId': exerciseId,
      'sets': sets,
      'reps': reps,
      'weight': weight,
      'orderIndex': orderIndex, // Оставляем orderIndex
      'imageUrl': imageUrl,
    };
  }
}
